### PR TITLE
feat(rest): endpoint to get obligations depending upon obligation level

### PR DESCRIPTION
**Pull request by rudra-superrr**
Issue: #2610 

Description: endpoint to get obligations depending upon obligation level (Possible values are: License, Project, Component, Organization).

### CODE REVIEW

**Comment by akshitjoshii** [comment]
Testing this PR

**Comment by akshitjoshii** [comment]
Test was successful.


image

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 1]
```diff
@@ -2488,6 +2488,61 @@ public ResponseEntity<HalResource> getLicenseObligations(Pageable pageable,
...
     }
 
+    @Operation(
+            description = "Get obligation data of project tab.",
+            tags = {"Project"}
+    )
+    @RequestMapping(value = PROJECTS_URL + "/{id}/obligation", method = RequestMethod.GET)
+    public ResponseEntity<HalResource> getObligations(Pageable pageable,
+                                                             @Parameter(description = "Project ID.") @PathVariable("id") String id,
+                                                             @Parameter(description = "Obligation Level")
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2498]: Please provide value set of supported values as `OblLevel` is a string.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java [hunk 2]
```diff
@@ -2488,6 +2488,61 @@ public ResponseEntity<HalResource> getLicenseObligations(Pageable pageable,
...
+                                                             @RequestParam(value = "obligationLevel", required = true) String OblLevel)
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java:2499]:
```suggestion
                                                             @RequestParam(value = "obligationLevel", required = true) String oblLevel)
```

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java [hunk 3]
```diff
@@ -437,6 +439,28 @@ public ObligationList getObligationData(String linkedObligationId, User user) th
         return sw360ProjectClient.getLinkedObligations(linkedObligationId, user);
     }
 
+    public Map<String, ObligationStatusInfo> setObligationsFromAdminSection(User user, Map<String, ObligationStatusInfo> obligationStatusMap,
+                                                                            Project project, String OblLevel) throws TException {
```
Comment (GMishx) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java:443]:
```suggestion
                                                                            Project project, String oblLevel) throws TException {
```

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by rudra-superrr** [comment]
Hi @GMishx , made the changes.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Pull request merged**